### DISCUSSION TITLE
How to configure proguard file?

### DISCUSSION THREAD

**Discussion by konradjurkowski** [sceneview/sceneview#586] [General]
I have applications everything works in debug mode, however when I build release then scene view does not work? Does anyone know what should I add to the proguard file?

**Comment by konradjurkowski** [comment]
Solved by adding it to proguard file

-keep class com.google.android.gms.location.** { *; }
-keep class com.google.android.gms.common.api.** { *; }
-keep class com.google.android.gms.tasks.** { *; }

**Comment by ThomasGorisse** [comment]
SceneView v4.0.0-rc.1 bundles its own ProGuard/R8 consumer rules, so you typically don't need to add anything manually.

If you do encounter obfuscation issues, add these to your `proguard-rules.pro`:

```
-keep class com.google.android.filament.** { *; }
-keep class io.github.sceneview.** { *; }
```

Make sure you're using R8 (Android's default optimizer) and that the SceneView dependency is pulled from Maven Central so the bundled consumer rules are applied automatically.

**Comment by ThomasGorisse** [comment]
SceneView v4 includes consumer ProGuard rules automatically — no manual configuration needed.

The rules are bundled in the AAR:
- `sceneview/consumer-rules.pro`
- `arsceneview/consumer-rules.pro`

If you still see issues with R8/minification, add:
```proguard
-keep class com.google.android.filament.** { *; }
-keep class io.github.sceneview.** { *; }
```

But this should not be necessary with the bundled rules.
